target/linux/realtek/dts/rtl8380_netgear_gigabit.dtsi

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rtl838x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "realtek,rtl838x-soc";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		mode {
+			label = "reset";
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio0 {
+	indirect-access-bus-id = <0>;
+};
+
+&spi0 {
+	status = "okay";
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0000000 0x0e0000>;
+				read-only;
+			};
+			partition@e0000 {
+				label = "u-boot-env";
+				reg = <0x00e0000 0x010000>;
+			};
+			partition@f0000 {
+				label = "sysinfo";
+				reg = <0x00f0000 0x010000>;
+				read-only;
+			};
+			partition@100000{
+				label = "jffs2_cfg";
+				reg = <0x0100000 0x100000>;
+				read-only;
+			};
+			partition@200000{
+				label = "jffs2_log";
+				reg = <0x0200000 0x100000>;
+			};
```

Comment:
This read-only atttibute usage seems arbitrary? AFAIK, OpenWrt do not access any of these by default.  Why is one of them read-only and not the other?

Reply:
I agree; in fact, I think that most of the partitions should be read-only by default, at least for now.  I've changed all but the main firmware partition to be read-only in the dts (and then only to facilitate sysupgrade).